api-builder.js

```diff
@@ -1,15 +1,18 @@
 module.exports = buildApi
 
 var routes =
-  { get: require('./endpoints/get')
-  , post: require('./endpoints/post')
-  , put: require('./endpoints/put')
-  , patch: require('./endpoints/patch')
-  , 'delete': require('./endpoints/delete')
-  }
+      { get: require('./endpoints/get')
+      , post: require('./endpoints/post')
+      , put: require('./endpoints/put')
+      , patch: require('./endpoints/patch')
+      , 'delete': require('./endpoints/delete')
+      }
+  , EventEmitter = require('events').EventEmitter
 
 function buildApi(service, urlRoot, router, logger, middleware, verbs) {
 
+  var apiBuilder = new EventEmitter()
```

Comment:
This is a weird name, the thing you return isn't an API builder, it's the API for whatever you built for. A better name might be simply `api` or `subRouter`?

Also, slightly nit-picky here but I find this gripe with a lot of modules where we use the revealing module pattern to export an instance of an `EventEmitter`, the error messages end up something like: `Error property: foo of Object event emitter is not a function`. To get around this I would suggest the following:

```js
var Emitter = require('events').EventEmitter
function SubRouter() { Emitter.call(this) }
SubRouter.prototype = Object.create(Emitter.prototype)
```

Then the object appears as `SubRouter`* instances in err/log messages rather than `EventEmitter` which is more helpful.

* I don't know if SubRouter is the best name, I just used it to flesh out this example.